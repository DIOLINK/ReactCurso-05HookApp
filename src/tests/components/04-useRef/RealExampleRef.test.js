import { shallow } from 'enzyme';
import RealExampleRef from '../../../components/04-useRef/RealExampleRef';

describe('Probando <RealExampleRef />', () => {
  it('Debe de mostrar el componente', () => {
    const wrapper = shallow(<RealExampleRef />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });
  it('Debe de mostrar <MultipleCustomHooks />', () => {
    const wrapper = shallow(<RealExampleRef />);
    const btn = wrapper.find('button');
    btn.simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });
});
