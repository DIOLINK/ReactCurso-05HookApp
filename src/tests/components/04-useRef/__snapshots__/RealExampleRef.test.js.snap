// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <RealExampleRef /> Debe de mostrar el componente 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
